File: weeks/2/3/solution.cpp
#include <algorithm>
#include <iomanip>
#include <iostream>
#include <utility>

using namespace std;

#define MAX_NUM_ROWS int(1e4+1)

pair<string, double> rows[MAX_NUM_ROWS];

void solve() {
  int numRows, index;
  cin >> numRows >> index;

  for (int i = 0; i < numRows; ++i) {
    cin >> rows[i].first >> rows[i].second;
  }

  sort(rows, rows+numRows);

  auto [name, height] = rows[index-1];
  cout << fixed << setprecision(2);
  cout << name << " " << height << '\n';
}

int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  solve();
}

File: weeks/2/3/solution.py
import sys
sys.setrecursionlimit(10**6)
readline = sys.stdin.readline

numRows, index = map(int, readline().split())
rows = []
for _ in range(numRows):
    row = readline().split()

    name = row[0]
    height = float(row[1])

    rows.append((name, height))

# 문자열을 사전순, 숫자를 오름차순 정렬하는 것이 기본값
rows.sort()

name, height = rows[index-1]
formattedHeight = "{:.2f}".format(height)
print(f'{name} {formattedHeight}')
